graphql/schema/schema.graphql

```diff
@@ -175,6 +175,9 @@ type Mutation {
   """Resets the o-counter for a scene to 0. Returns the new value"""
   sceneResetO(id: ID!): Int!
 
+  """Update resume time as well as play count and play duration"""
+  sceneSaveActivity(id: ID!, resume_time: Float!, playDuration: Float!): Int!
```

Comment:
I think we should have the external API as follows:

```
"""Sets the resume time point (if provided) and adds the provided duration to the scene's play duration"""
sceneSaveActivity(id: ID!, resume_time: Float, playDurationToAdd: Float): Bool!
```
This allows a client to set the resume time, and/or add to the scene's play duration. It should _not_ increment the play count.

Propose adding a _separate_ interface for incrementing the watch count.

```
"""Increments the watch count for the scene. Returns the new watch count value."""
sceneIncrementWatchCount(id: ID!): Int!
```

Other operations for watch count (setting/resetting or decrementing) can be done via the `sceneUpdate` interface.

Reply:
I just saw your comments after I had already pushed changes to improve the existing approach. I can rework the code work the way you described above.